test/utils/p4rtsimulator/simulator.go

```diff
@@ -7,18 +7,133 @@ package p4rtsimulator
 import (
 	"context"
 	"github.com/onosproject/helmit/pkg/helm"
+	"github.com/onosproject/helmit/pkg/kubernetes"
 	"github.com/onosproject/helmit/pkg/util/random"
+	"github.com/onosproject/onos-api/go/onos/topo"
+	toposdk "github.com/onosproject/onos-ric-sdk-go/pkg/topo"
 	"github.com/onosproject/onos-test/pkg/onostest"
 	"github.com/stretchr/testify/assert"
+	"strconv"
 	"testing"
 	"time"
 )
 
 const (
 	// Maximum time for an entire test to complete
 	defaultTestTimeout = 3 * time.Minute
+
+	// SimulatorTargetType type for simulated target
+	SimulatorTargetType = "devicesim"
+	// SimulatorTargetVersion default version for the simulated target
+	SimulatorTargetVersion = "1.0.0"
+
+	defaultTimeout = time.Second * 30
 )
 
+// NewSimulatorTargetEntity creates a topo entity for a device simulator target
+func NewSimulatorTargetEntity(ctx context.Context, simulator *helm.HelmRelease, targetType string, targetVersion string) (*topo.Object, error) {
+	simulatorClient := kubernetes.NewForReleaseOrDie(simulator)
+	services, err := simulatorClient.CoreV1().Services().List(ctx)
+	if err != nil {
+		return nil, err
+	}
+	service := services[0]
+	return NewTargetEntity(simulator.Name(), targetType, targetVersion, service.Ports()[0].Address(true))
+}
+
+// NewTargetEntity creates a topo entity with the specified target name, type, version and service address
+func NewTargetEntity(name string, targetType string, targetVersion string, serviceAddress string) (*topo.Object, error) {
+	o := topo.Object{
+		ID:   topo.ID(name),
+		Type: topo.Object_ENTITY,
+		Obj: &topo.Object_Entity{
+			Entity: &topo.Entity{
+				KindID: topo.ID(targetType),
+			},
+		},
+	}
+
+	if err := o.SetAspect(&topo.TLSOptions{Insecure: true, Plain: true}); err != nil {
+		return nil, err
+	}
+
+	timeout := defaultTimeout
+	if err := o.SetAspect(&topo.Configurable{
+		Type:                 targetType,
+		Address:              serviceAddress,
+		Version:              targetVersion,
+		Timeout:              &timeout,
+		ValidateCapabilities: true,
+	}); err != nil {
+		return nil, err
+	}
+
+	return &o, nil
+}
+
+// NewSwitchEntity creates a switch entity with specified information
+func NewSwitchEntity(name string, version string, serviceAddress string, servicePort string, deviceID uint64, modelID string, role string) (*topo.Object, error) {
+	o := topo.Object{
+		ID:   topo.ID(name),
+		Type: topo.Object_ENTITY,
+		Obj: &topo.Object_Entity{
+			Entity: &topo.Entity{
+				KindID: topo.ID("switch"),
```

Comment:
You should use predefined kind IDs from topo api https://github.com/onosproject/onos-api/blob/master/go/onos/topo/topo.go#L58